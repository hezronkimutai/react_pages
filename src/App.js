import React from 'react'
import Example from './demo/Example';

const App = () => {
  return (
    <Example />
  )
}

export default App
