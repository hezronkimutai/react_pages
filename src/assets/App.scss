%fullDimensions {
  width: 100%;
  margin: auto;
}
%flex-row {
  display: flex;
  flex-direction: row;
}
html {
  position: fixed;
  @extend %fullDimensions;
  body {
    height: 60%;
    width: 100%;
    margin: auto;
    #app {
      @extend %fullDimensions;
      .container {
        margin: auto;
        margin-top: 10%;
        width: 70%;
        background-color: #999999;
        height: 100%;
        h1 {
          width: 100%;
          text-align: center;
        }
        .pagination-container {
          display: flex;
          flex-direction: column;
          .oneItem {
            @extend %flex-row;
            height: 15%;
            div {
              padding: 20px;
            }
          }
          .pagination-buttons {
            display: flex;
            flex-direction: row;
            padding: 20px 50px 20px 0px;
          }
          
          .pagination-buttons button {
            height: 36px;
            cursor: pointer;
            font-size: 16px;
          }
          .pages {
            display: flex;
            flex-direction: row;
          }
          
          .pages button {
            border-bottom: 1px solid #d5d4d0;
            border-top: 1px solid #d5d4d0;
          }
          
          button#prev {
            border-bottom: 1px solid #d5d4d0;
            border-top: 1px solid #d5d4d0;
            border-bottom-left-radius: 5px;
            border-top-left-radius: 5px;
          }
          
          button#next {
            border-bottom: 1px solid #d5d4d0;
            border-top: 1px solid #d5d4d0;
            border-bottom-right-radius: 5px;
            border-top-right-radius: 5px;
          }
        }
      }
    }
  }
}
