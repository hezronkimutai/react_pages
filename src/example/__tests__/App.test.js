import React from 'react'
import { shallow, mount } from 'enzyme'
import App from '../App'

describe('App test.', () => {
  test('Buttons and snapshot when props is provided', () => {
    const wrapper = mount(<App />)
    wrapper
      .find('button')
      .at(0)
      .simulate('click')
    wrapper
      .find('button')
      .at(1)
      .simulate('click')
    wrapper
      .find('button')
      .at(2)
      .simulate('click')
    wrapper
      .find('button')
      .at(3)
      .simulate('click')
    expect(wrapper).toMatchSnapshot()
  })
  test('Buttons and snapshot when props is provided', () => {
    const wrapper = mount(<App />)
    wrapper
      .find('button')
      .at(3)
      .simulate('click')
    wrapper
      .find('button')
      .at(0)
      .simulate('click')
    expect(wrapper).toMatchSnapshot()
  })
})
