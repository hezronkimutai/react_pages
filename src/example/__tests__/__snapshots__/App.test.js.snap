// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test. Buttons and snapshot when props is provided 1`] = `ReactWrapper {}`;

exports[`App test. Buttons and snapshot when props is provided 2`] = `ReactWrapper {}`;
